examples/lin_algebra/gmres.cpp

```diff
@@ -0,0 +1,274 @@
+      H[h_idx] = norm_h_idx;
+      v.col(i+1) = (1.0/H[h_idx])*w;
+      for (int k = 0; k < i; k++)
+        ApplyPlaneRotation<T>( H[k+i*(R+1)], H[(k+1)+i*(R+1)], cs[k], sn[k]);
+      GeneratePlaneRotation<T>( H[i+i*(R+1)], H[(i+1)+i*(R+1)], cs[i], sn[i]);
+      ApplyPlaneRotation<T>( H[i+i*(R+1)], H[(i+1)+i*(R+1)], cs[i], sn[i]);
+      ApplyPlaneRotation<T>( s[i], s[(i+1)], cs[i], sn[i]);
+      if (abs(s[i+1])/normb < tol){
+        Update<T>(x, i, H, R, s, v);
+        return iter;
+      }    
+    }
+    Update<T>(x, R-1, H, R, s, v); 
+    r = b -matmul(A,x);
+    beta = norm(r);  
+     
+    if ((beta / normb) < tol) {
+      return iter;
+    }
+  }
+ return iter;
+}
+template <class T>
+static void solve_demo(dtype dt, info_solver &solve_info) { 
+  /*
+  solving laplace equation
+  f_xx + f_yy = 0 : x in (0,1), y in (0,1)
+  with the below Dirichlet boundary condition
+  f(x,0) = 4 - x^2,
+  f(x,1) = 3 -x^2,
+  f(0,y) = 4 - y^2,
+  f(1,y) = 3 -y^2.
+  The exact solution is f(x,y) = 4 - x^2 + y^2
+  The domain is discretized into nxn grid points
+  with a central difference scheme, the discretized equation is as folows:
+  (f(x+dx,y) + f(x-dx,y) + f(x,y+dy) + f(x,y-dy) - 4f(x,y))/dx^2 = 0
+  where dx=dy=1/(n-1).
+  This results in a 5 diagonal matrix
+  For instance, for n =3, the matrix is 9x9 will look like this:
+   _                                                    _
+  | 1.00  0.00  0.00  0.00  0.00  0.00  0.00  0.00  0.00 |  
+  | 0.00  1.00  0.00  0.00  0.00  0.00  0.00  0.00  0.00 | 
+  | 0.00  0.00  1.00  0.00  0.00  0.00  0.00  0.00  0.00 | 
+  | 0.00  0.00  0.00  1.00  0.00  0.00  0.00  0.00  0.00 | 
+  | 0.00  -.25  0.00  -.25  1.00  -.25  0.00  -.25  0.00 | 
+  | 0.00  0.00  0.00  0.00  0.00  1.00  0.00  0.00  0.00 | 
+  | 0.00  0.00  0.00  0.00  0.00  0.00  1.00  0.00  0.00 | 
+  | 0.00  0.00  0.00  0.00  0.00  0.00  0.00  1.00  0.00 | 
+  |_0.00  0.00  0.00  0.00  0.00  0.00  0.00  0.00  1.00_|
+  */
+  printf("Solving 2D Laplace equation with GMRES method\n");
+  for (int n = 20; n<400; n*=2){
+
+    int m = n*n;
+    std::vector<T> b_host(m);
+    std::vector<T> x_exact(m);
+    std::vector<T> A_host;
+    std::vector<int> row_indices;
+    std::vector<int> col_indices;
+  
+    printf("matrix size %dx %d: with %d nonzero elements.\n",n,n,(n-2)*(n-2)*5+4*(n-1));
+    
+    T dx = 1./(n-1);
+    // generate the matrix in the COO format.
+    for (int i = 0; i < n; i++) {
+      for (int j = 0; j < n; j++) {
+        int idx = i * n + j;
+        int row_id = idx;
+        T xl = i*dx;
+        T yl = j*dx;
+        x_exact[idx] =  ((xl-.5)*(xl-0.5) - (yl-0.5)*(yl-0.5));
+        if (i == 0 || i == n - 1 || j == 0 || j == n - 1) {
+          
+          A_host.push_back(1);
+          row_indices.push_back(row_id);
+          col_indices.push_back(row_id);
+        }
+        else
+        {   
+          row_indices.push_back(row_id);
+          col_indices.push_back(row_id-n);
+          A_host.push_back(-1.0);
```

Comment:
-1.0 should be -0.25 and 4.0 should be 1.0 here to be consistent with the example matrix given above.